### PR TITLE
Enable optimizer in aleth-interpreter

### PR DESCRIPTION
**Pull request by gumb0**
Similar to how LegacyVM enables it https://github.com/ethereum/aleth/blob/39a9e4323b961d7b12cace37f5ed339f4b1abff5/libevm/CMakeLists.txt#L22-L24

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@
 - Fixed: [#5664](https://github.com/ethereum/aleth/pull/5664) Behavior in corner case tests about touching empty Precompiles now agrees with geth's results.
 - Fixed: [#5662](https://github.com/ethereum/aleth/pull/5662) Correct depth value when aleth-interpreter invokes `evmc_host_interface::call` callback.
 - Fixed: [#5666](https://github.com/ethereum/aleth/pull/5666) aleth-interpreter returns `EVMC_INVALID_INSTRUCTION` when `INVALID` opcode is encountered and `EVMC_UNKNOWN_INSTRUCTION` for undefined opcodes.
+- Fixed: [#5678](https://github.com/ethereum/aleth/pull/5678) Enable optimizer in aleth-interpreter by default.
```
Comment (chfast) [inline CHANGELOG.md]: Rather make it `Changed`.
